File: projetoJava/src/Empresa.java
import java.util.ArrayList;
import java.util.List;


class Empresa {
    private String nome;
    private List<Empregado> empregados;

    public Empresa(String nome) {
        this.nome = nome;
        this.empregados = new ArrayList<>();
    }

    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    public String infoEmpresa() {
        StringBuilder info = new StringBuilder("Empresa: " + nome + "\nEmpregados:\n");
        for (Empregado empregado : empregados) {
            info.append("- ").append(empregado.infoEmpregado()).append("\n");
        }
        return info.toString();
    }
}

File: projetoGo/empresa.go
package main

import "fmt"

type Empresa struct {
	nome       string
	empregados []*Empregado
}

func (e *Empresa) AdicionarEmpregado(empregado *Empregado) {
	e.empregados = append(e.empregados, empregado)
}

func (e Empresa) InfoEmpresa() string {
	info := fmt.Sprintf("Empresa: %s\nEmpregados:\n", e.nome)
	for _, empregado := range e.empregados {
		info += fmt.Sprintf("- %s\n", empregado.InfoEmpregado())
	}
	return info
}
